tiledb/cf/core.py

```diff
@@ -462,6 +406,71 @@ def meta(self) -> Optional[tiledb.Metadata]:
             return None
         return self._metadata_array.meta
 
+    def open_array(
+        self,
+        array: Optional[str] = None,
+        attr: Optional[str] = None,
+        mode: str = None,
+    ) -> tiledb.Array:
+        if mode is None:
+            mode = self._mode
+        if array is None and attr is None:
+            raise ValueError(
+                "Cannot open array. Either an array or attribute must be specified."
+            )
+        if array is None:
+            array_names = self._group_schema.arrays_with_attr(attr)
+            if array_names is None:
+                raise KeyError(f"No attribute with name '{attr}' found.")
+            if len(array_names) > 1:
+                raise ValueError(
+                    f"The array must be specified when opening an attribute that "
+                    f"exists in multiple arrays in a group. Arrays with attribute "
+                    f"'{attr}' include: {array_names}."
+                )
+            array = array_names[0]
+        tiledb_array = tiledb.open(
+            self._array_uris[array],
+            mode=self._mode,
+            key=_get_array_key(self._key, array),
+            attr=attr,
+            config=None,
+            timestamp=self._timestamp,
+            ctx=self._ctx,
+        )
+        array_key = (array, attr)
+        self._open_arrays[array_key] = tiledb_array
+        return tiledb_array
+
+    def close_array(self, array: Optional[str] = None, attr: Optional[str] = None):
+        if array is None and attr is None:
+            raise ValueError(
+                "Cannot open array. Either an array or attribute must be specified."
+            )
+        if array is None:
+            array_names = self._group_schema.arrays_with_attr(attr)
+            if array_names is None:
+                raise KeyError(f"No attribute with name '{attr}' found.")
+            if len(array_names) > 1:
+                raise ValueError(
+                    f"The array must be specified when opening an attribute that "
+                    f"exists in multiple arrays in a group. Arrays with attribute "
+                    f"'{attr}' include: {array_names}."
+                )
+            array = array_names[0]
+        array_key = (array, attr)
+        try:
+            tiledb_array = self._open_arrays[array_key]
+            del self._open_arrays[array_key]
```

Comment:
```suggestion
            tiledb_array = self._open_arrays.pop(array_key)
```
You can combine the get & delete into a single pop.